### PR TITLE
init: add random chain-id generator

### LINKED ISSUE #681: It returns error when "--chain-id" was set empty during "ethermintd init"
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]
It says that "genesis file chain-id, if left blank will be randomly created" in `ethermintd init -h`
image

commit：9659b81

__Steps to reproduce:__

It set empty for `--chaind-id` :
```shell
 $ ethermintd init alice
```

__Expected behavior:__ [What you expected to happen]

__Actual behavior:__ [What actually happened]
And it returns an error:
```
ERROR: invalid chain-id format: 
```
__Additional info:__ [Include gist of relevant config, logs, etc.]

### PR DESCRIPTION
**Pull request by helder-moreira**
Closes: #681

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by alessio — APPROVED** [review]

File: types/chain_id.go [hunk 1]
```diff
@@ -46,3 +47,8 @@ func ParseChainID(chainID string) (*big.Int, error) {
 
 	return chainIDInt, nil
 }
+
+// GenerateRandomChainID returns a random chain-id in the valid format.
+func GenerateRandomChainID() string {
+	return fmt.Sprintf("testchain-%d", 10+tmrand.Intn(10000))
```
Comment (fedekunze) [inline types/chain_id.go]:
```suggestion
	return fmt.Sprintf("ethermint-%d", 10+tmrand.Intn(10000))
```

File: client/config.go [hunk 2]
```diff
@@ -61,3 +61,23 @@ func ValidateChainID(baseCmd *cobra.Command) *cobra.Command {
...
+	// Copy base run command to be used after chain verification
+	baseRunE := baseCmd.RunE
+
+	// Function to replace command's RunE function
+	generateFn := func(cmd *cobra.Command, args []string) error {
+		chainID := viper.GetString(flags.FlagChainID)
+
+		if chainID == "" {
+			viper.Set(flags.FlagChainID, ethermint.GenerateRandomChainID())
+		}
```
Comment (fedekunze) [inline client/config.go:77]: I'd prefer to add this logic to the `ValidateChainID` function

**Review by fedekunze — COMMENTED** [review]
Minor comment

File: client/config.go (hunk 2 above)
Comment (alessio) [reply client/config.go:77]: But then `ValidateChainID` would not just validate

**Review by fedekunze — APPROVED** [review]

**Pull request merged**